lib/jobs/message_latency_report.ex

```diff
@@ -210,13 +198,27 @@ defmodule Jobs.MessageLatencyReport do
     id
   end
 
-  defp get_percentile_row(percentile_index, rows_with_index) do
-    {percentile_row, _index} =
-      Enum.find(rows_with_index, {[], nil}, fn {_row, index} ->
-        index == percentile_index
-      end)
+  defp get_percentiles(rows_with_index) do
+    num_rows = Enum.count(rows_with_index)
+    percentile_95_index = calculate_percentile_index(num_rows, @percentile_95)
+    percentile_99_index = calculate_percentile_index(num_rows, @percentile_99)
 
-    percentile_row
+    {Enum.reduce_while(rows_with_index, {[], []}, fn {current_row, index},
```

Comment:
I think `nil` would be a better guard value than `[]`, since these don't end up as lists, right?

Reply:
Oh great catch, yeah they used to return as lists before the refactor. The actual values should be floats though so I think I'll just opt to use two zeros